runway/utils.py

```diff
@@ -23,6 +23,19 @@
         r'(?::\d+)?' # optional port
         r'(?:/?|[/?]\S+)$', re.IGNORECASE)
 
+def validate_post_request_body_is_json(f):
+    @functools.wraps(f)
+    def wrapped(*args, **kwargs):
+        json = get_json_or_none_if_invalid(request)
+        if json is not None:
+            return f(*args, **kwargs)
+        else:
+            err_msg = 'The body of all POST requests must contain JSON'
+            return jsonify(dict(error=err_msg)), 400
+    return wrapped
+
+def get_json_or_none_if_invalid(request):
+    return request.get_json(force=True, silent=True)
```

Comment:
This is a thin wrapper around Flask's `request.get_json()` but it ensures that we are calling it such that it will sniff JSON even if the `content-type: application/json` MIME type is not set, and will return `None` if the sniff fails instead of throwing an error.

We can then use this util function elsewhere to make sure that we are parsing JSON from Flask response objects in a consistent way.